themes/src/main/resources/theme/base/login/login-config-totp.ftl

```diff
@@ -58,7 +58,12 @@
             <#if mode??><input type="hidden" id="mode" name="mode" value="${mode}"/></#if>
         </div>
 
-        <input class="${properties.kcButtonClass!} ${properties.kcButtonPrimaryClass!} ${properties.kcButtonBlockClass!} ${properties.kcButtonLargeClass!}" type="submit" value="${msg("doSubmit")}"/>
+        <#if isAppInitiatedAction??>
+        <input class="${properties.kcButtonClass!} ${properties.kcButtonPrimaryClass!} ${properties.kcButtonLargeClass!}" type="submit" value="${msg("doSubmit")}" />
```

Comment:
Why is the submit button duplicated? Isn't it better to just wrap the cancel button with isAppInitated?

Reply:
It is for formatting purposes.   Today, the submit button takes up the entire width of the screen (which is probably wrong).  But I kept it the way it has always been when you aren't doing AIA.